lib/taskbook.js

```diff
@@ -325,8 +325,50 @@ class Taskbook {
     render.successCreate(task);
   }
 
+  _flattenDown(array, result) {
+    for (let i = 0; i < array.length; i++) {
+      const value = array[i];
+
+      if (Array.isArray(value)) {
+        this._flattenDown(value, result);
+      } else {
+        result.push(value);
+      }
+    }
+
+    return result;
+  }
+
+  _generateNumbers(value) {
+    let _start = value[0] || null;
+    let _end = value[1] || null;
+    const _arr = [];
+
+    if (!_start || !_end) {
+      return 0;
+    }
+
+    if (_start === _end) {
+      return _start;
+    }
+
+    if (!parseInt(_start, 10) < parseInt(_end, 10)) {
+      [_start, _end] = [_end, _start];
+    }
+
+    for (let i = parseInt(_start, 10); i <= parseInt(_end, 10); i++) {
+      _arr.push(i);
+    }
+
+    return _arr;
+  }
+
+  _produceNumbers(values) {
+    return [...new Set(this._flattenDown(values.map(value => value.indexOf('-') > -1 ? this._generateNumbers(value.split('-')) : value), []))];
```

Comment:
I personally think that this line looks more complex than it really should be in the first look.
Why not improve this function to actually iterate in the values and generate a new array directly?
This way you will not need to flatten the array afterwards.